modules/underdogmediaBidAdapter.js

```diff
@@ -1,112 +1,116 @@
+import * as utils from 'src/utils';
 import { config } from 'src/config';
-var bidfactory = require('src/bidfactory.js');
-var bidmanager = require('src/bidmanager.js');
-var adloader = require('src/adloader.js');
-var utils = require('src/utils.js');
-var adaptermanager = require('src/adaptermanager');
-
-function UnderdogMediaAdapter() {
-  const UDM_ADAPTER_VERSION = '1.0.0';
-  var getJsStaticUrl = window.location.protocol + '//udmserve.net/udm/img.fetch?tid=1;dt=9;callback=$$PREBID_GLOBAL$$.handleUnderdogMediaCB;';
-  var bidParams = {};
-
-  function _callBids(params) {
-    bidParams = params;
+import { registerBidder } from 'src/adapters/bidderFactory';
+const BIDDER_CODE = 'underdogmedia';
+const UDM_ADAPTER_VERSION = '1.0';
+
+export const spec = {
+  code: BIDDER_CODE,
+  bidParams: [],
+
+  isBidRequestValid: function (bid) {
+    return !!((bid.params && bid.params.siteId) && (bid.sizes && bid.sizes.length > 0));
+  },
+
+  buildRequests: function (validBidRequests) {
+    spec.bidParams = validBidRequests;
     var sizes = [];
     var siteId = 0;
 
-    bidParams.bids.forEach(bidParam => {
+    spec.bidParams.forEach(bidParam => {
       sizes = utils.flatten(sizes, utils.parseSizesInput(bidParam.sizes));
       siteId = bidParam.params.siteId;
     });
-    adloader.loadScript(getJsStaticUrl + 'sid=' + siteId + ';sizes=' + sizes.join(','), null, false);
-  }
 
-  function _callback(response) {
-    var mids = response.mids;
-    bidParams.bids.forEach(bidParam => {
-      var filled = false;
-      mids.forEach(mid => {
+    return {
+      method: 'GET',
+      url: `${window.location.protocol}//udmserve.net/udm/img.fetch`,
+      data: `tid=1;dt=10;sid=${siteId};sizes=${sizes.join(',')}`,
```

Comment:
More params can be added to request object here. So you can add `bidParams` to bidRequest and access that in `interpretResponse` method. 
By doing this you can remove `bidParams` line 15.